pytorch_lightning/plugins/training_type/deepspeed.py

```diff
@@ -517,18 +521,10 @@ def optimizer_step(self, optimizer: torch.optim.Optimizer, lambda_closure: Calla
         # internally, the engine has a reference to the optimizer already.
         self.model.step(**kwargs)
 
-    def _handle_gradient_accumulation_steps(self):
-        """This functions overrides the trainer.accumulation_scheduler to generate ``accumulate_grad_batches=1``.
-
-        Therefore, ``optimizer_step`` will be called on every batches seen so DeepSpeed Engine handles the gradient
-        accumulation logic internally.
-        """
-        if self.config.get("gradient_accumulation_steps") > 1:
-            self._original_accumulate_grad_batches = self.lightning_module.trainer.accumulate_grad_batches
-            # todo (tchaton) Add support for accumulate_grad_batches being a dictionary.
-            self.lightning_module.trainer.accumulation_scheduler = GradientAccumulationScheduler({0: 1})
-        else:
-            self._original_accumulate_grad_batches = None
+    @property
+    def handles_accumulate_grad_batches(self) -> bool:
```

Comment:
"accumulate_grad_batches" was not a great name from the beginning imo. do we want to take this opportunity and choose the right name? like `handles_gradient_accumulation` for example?